src/state-visualizer/visualizer.py

```diff
@@ -0,0 +1,141 @@
+import graphviz
+import time
+from PyQt5.QtWidgets import *
+from PyQt5.QtCore import *
+from PyQt5.QtGui import QPainter
+from PyQt5.QtSvg import QSvgRenderer
+
+import rospy
+from smach_msgs.msg import SmachContainerStatus,SmachContainerStructure
+from threading import Lock
+from dataclasses import dataclass, field
+from typing import Optional
+
+@dataclass
+class State:
+    name : str
+    children : list #of states
+
+@dataclass
+class StateMachine:
+    states : dict  = field(default_factory=dict)#maps string name to State
+    structure : Optional[SmachContainerStructure] = None
+    mutex : Lock = Lock()
```

Comment:
I don't think it makes sense for a class with a mutex to be a dataclass?

Reply:
not sure I understand why it doesn't make sense. Are you proposing I just make it a normal class and do something like `self.mutex = Lock()`?

Reply:
I suppose it is fine if you are only using it to generate repr and init

My point more was a dataclass is supposed to be for data only. A mutex implies a shared state